vmcnet/train/runners.py

```diff
@@ -607,13 +640,26 @@ def run_molecule() -> None:
 
     # need to check for local_energy.txt because when config.eval.nepochs=0 the file is
     # not created regardless of config.eval.record_local_energies
-    local_es_were_recorded = os.path.exists(
-        os.path.join(eval_logdir, "local_energies.txt")
-    )
+    local_energies_filepath = os.path.join(eval_logdir, "local_energies.txt")
+    local_es_were_recorded = os.path.exists(local_energies_filepath)
     if config.eval.record_local_energies and local_es_were_recorded:
-        local_energies_filepath = os.path.join(eval_logdir, "local_energies.txt")
-        _compute_and_save_energy_statistics(
-            local_energies_filepath, eval_logdir, "statistics"
+        _compute_and_save_statistics(
+            local_energies_filepath, eval_logdir, "energy_statistics"
+        )
+
+    local_spin_exchanges_filepath = os.path.join(
+        eval_logdir, "local_spin_exchanges.txt"
+    )
+    local_spin_exchanges_were_recorded = os.path.exists(local_spin_exchanges_filepath)
+    if config.eval.record_local_spin_exchanges and local_spin_exchanges_were_recorded:
+        spin_squared_offset = 0.25 * (nelec[0] - nelec[1]) ** 2 + 0.5 * (
```

Comment:
This offset stuff seems a bit hacky. Could we instead expose a function in spin.py that adds this offset to the exchange term, and use that as our local_spin_exchange_fn from the get go? Preferably also within spin.py we would only define this offset once and use it both in the spin square expectation and in this offsetted local spin exchange function.

Reply:
The main reason not to add the offset to the exchange term is that it's more expensive (nwalkers additions instead of 1). It would probably be cleaner though, certainly. Let me think a bit about how to make it cleaner

Reply:
(I'm talking about the expensiveness during training, not this particular use. Probably fine to do nwalkers additions once here, but would like to avoid doing it repeatedly during training)

Reply:
Mm I see. Idk if we need to care about this level of optimization though, we're doing a lot of work per walker to evaluate the network, calculate gradients, etc, so I feel like one extra addition op would be negligible?

Reply:
True, probably unnecessary for me to worry about it in terms of the computational cost. Maaaaybe there's something to be worried about in terms of floating point error being magnified but it seems like I'm just adding the offset to the local quantity for this statistics computation anyway so that's not exactly fixed in the current implementation where it matters. I'll revisit this code